### PR TITLE
Replace Sentry capture call in SecurityEventLogger with logger call

### PR DESCRIPTION
**Pull request by jhchabran**
This PR remove a combination of a log15 call and explicit capture in favour of our new logging api and implicitly capturing the error.

## Test plan

Rather trivial change, logging api has been tested exhaustively already.

### CODE REVIEW

**Comment by jhchabran** [comment]
Current dependencies on/for this PR:
* main
  * **PR #37209** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/37209" target="_blank">Graphite</a> 
    * **PR #37212** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/37212" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/37212?utm_source=stack-comment).

File: internal/database/security_event_logs.go [hunk 1]
```diff
@@ -67,12 +65,14 @@ type SecurityEventLogsStore interface {
...
 type securityEventLogsStore struct {
 	*basestore.Store
+	logger log.Logger
 }
 
 // SecurityEventLogsWith instantiates and returns a new SecurityEventLogsStore
 // using the other store handle.
 func SecurityEventLogsWith(other basestore.ShareableStore) SecurityEventLogsStore {
-	return &securityEventLogsStore{Store: basestore.NewWithHandle(other.Handle())}
+	logger := log.Scoped("security_events", "Security events logs")
```
Comment (bobheadxi) [inline internal/database/security_event_logs.go:74]:
```suggestion
	logger := log.Scoped("SecurityEvents", "Security events")
```

File: internal/database/security_event_logs.go [hunk 2]
```diff
@@ -108,9 +108,6 @@ func (s *securityEventLogsStore) LogEvent(ctx context.Context, e *SecurityEvent)
 
 	if err := s.Insert(ctx, e); err != nil {
 		j, _ := json.Marshal(e)
-		log15.Error(string(e.Name), "event", string(j), "traceID", trace.ID(ctx), "error", err)
-		// We want to capture in sentry as it includes a stack trace which will allow us
-		// to track down the root cause.
-		sentry.CaptureError(err, map[string]string{})
+		s.logger.Error(string(e.Name), log.String("event", string(j)), log.String("traceID", trace.ID(ctx)), log.Error(err))
```
Comment (bobheadxi) [inline internal/database/security_event_logs.go:111]: I just notice we have a trace ID, this should use our `WithTrace` logging: https://docs.sourcegraph.com/dev/how-to/add_logging#traced-sub-loggers (sorry I missed this on the first review!)

**Review by bobheadxi — CHANGES_REQUESTED** [review]

File: internal/database/security_event_logs.go (hunk 2 above)
Comment (jhchabran) [reply internal/database/security_event_logs.go:111]: Oh nice catch! (np, I'll take a late catch before the merge any day over a post merge one 😆👍)

**Review by bobheadxi — APPROVED** [review]
beautiful :)

**Pull request merged**